roles/server/tasks/RedHat.yml

```
@@ -18,15 +18,14 @@
     state: true
     persistent: true
 
-- name: "Get service facts."
-  ansible.builtin.service_facts:
-  register: services_state
-
 - name: "Open port 80 for httpd."
   become: true
   ansible.posix.firewalld:
     service: http
     permanent: true
     immediate: true
     state: enabled
-  when: services_state.ansible_facts.services['firewalld.service']['state'] == 'running'
+  with_items:
+    - "{{ checkmk_server_ports }}"
+  when: checkmk_server_change_firewall | bool
+  ignore_errors: true
```

This feels dangerous, why is this necessary?

---

I did this so it wouldn't halt the playbook for the host if firewalld isn't running on the host. And I did this instead of the previous check since you're more likely to notice a fail rather than it skipping the host. But the firewall not being configured doesn't really block any other tasks, so it doesn't really need to stop. But if you think it should just stop the run after that failing and you should just know to set it to false for hosts where it's not running/failed before that seems okay too, I can remove the ignore errors if you want.

---

Jup, I rather have a task fail once too often than the other way around.